src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -1163,8 +1163,20 @@ TripleComponent::Iri Visitor::visit(Parser::IriContext* ctx) {
 }
 
 // ____________________________________________________________________________________
-string Visitor::visit(Parser::IrirefContext* ctx) {
-  return RdfEscaping::unescapeIriref(ctx->getText());
+string Visitor::visit(Parser::IrirefContext* ctx) const {
+  auto string = ctx->getText();
+  auto removeIriBrackets = [](std::string_view string) {
+    AD_CORRECTNESS_CHECK(string.starts_with('<'));
+    AD_CORRECTNESS_CHECK(string.ends_with('>'));
+    return string.substr(1, string.size() - 2);
+  };
+  // Handle relative IRIs, technically relative IRIs are not allowed when no
+  // BASE declaration is present, but we use them a lot for unit tests.
+  if (!baseIri_.empty() && !isAbsoluteIri(string)) {
+    return absl::StrCat("<", removeIriBrackets(baseIri_),
+                        removeIriBrackets(string), ">");
+  }
```

Comment:
That doesn't quite handle all cases, there are local and global references (don't remember the exact terminology) but @hannahbast has handled all those cases for the TURTLE parser. Can you please have a look at the methods for handling BASE correctly in the `ad_utility::triple_component::Iri` class, and reuse the code from there?